### PR TITLE
Add 'disableSuggestions' to UserPreferences

### LINKED ISSUE #23177: Add new option to suppress suggessions
Suggestions are automatically queried for, and editors (VSCode) for instance has no way of opting out of it. we need a new configuration option to tell the server to not query for them.

### PR DESCRIPTION
**Pull request by andy-ms**
Fixes #23177

### CODE REVIEW

**Comment by andy-ms** [comment]
@sheetalkamat If I comment out the new test, the failures in the other tests go away -- any idea how these are effecting each other?

**Comment by sheetalkamat** [comment]
looking

**Comment by sheetalkamat** [comment]
@andy-ms this is i think because of https://github.com/Microsoft/TypeScript/blob/master/src/server/editorServices.ts#L1849
It modifies host.preferences which could be defaultPreferences object which gets modified (https://github.com/Microsoft/TypeScript/blob/master/src/server/editorServices.ts#L452)

**Comment by andy-ms** [comment]
Ugh, every property in `UserPreferences` is readonly, but readonly modifiers truly are a joke. #13002
Thanks for debugging that!

**Comment by andy-ms** [comment]
@sheetalkamat Good to go?

File: src/server/scriptInfo.ts [hunk 1]
```diff
@@ -398,14 +398,14 @@ namespace ts.server {
                 if (!this.formatSettings) {
                     this.formatSettings = getDefaultFormatCodeSettings(this.host);
```
Comment (sheetalkamat) [inline src/server/scriptInfo.ts]: This function returns a copy, so it is ok to modify .. So no need to clone unlike defaultPreferences... I am not sure how many times we call these configure function to say if its ok to clone the object (esp with such a big set of properties)

File: src/server/scriptInfo.ts [hunk 2]
```diff
@@ -397,15 +397,18 @@ namespace ts.server {
             if (formatSettings) {
                 if (!this.formatSettings) {
                     this.formatSettings = getDefaultFormatCodeSettings(this.host);
+                    assign(this.formatSettings, formatSettings);
+                }
+                else {
+                    this.formatSettings = mergeMapLikes(this.formatSettings, formatSettings);
```
Comment (mhegazy) [inline src/server/scriptInfo.ts]: is not this just: 
```ts
this.formatSettings = { ... this.formatSettings, ...formatSettings};
```

File: src/server/scriptInfo.ts (hunk 1 above)
Comment (andy-ms) [reply src/server/scriptInfo.ts]: Done.

**Comment by andy-ms** [comment]
CC @mjbvz This should be added as a vscode setting.

**Pull request merged**